test-framework/junit5/src/main/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrderer.java

```diff
@@ -69,31 +71,57 @@ public class QuarkusTestProfileAwareClassOrderer implements ClassOrderer {
+        Config config = ConfigProvider.getConfig();
+        this.prefixQuarkusTest = config.getOptionalValue(CFGKEY_ORDER_PREFIX_QUARKUS_TEST, String.class)
+                .orElse(DEFAULT_ORDER_PREFIX_QUARKUS_TEST);
+        this.prefixQuarkusTestWithProfile = config.getOptionalValue(CFGKEY_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE, String.class)
+                .orElse(DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE);
+        this.prefixQuarkusTestWithRestrictedResource = config
+                .getOptionalValue(CFGKEY_ORDER_PREFIX_QUARKUS_TEST_WITH_RESTRICTED_RES, String.class)
+                .orElse(DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_RESTRICTED_RES);
+        this.prefixNonQuarkusTest = config.getOptionalValue(CFGKEY_ORDER_PREFIX_NON_QUARKUS_TEST, String.class)
+                .orElse(DEFAULT_ORDER_PREFIX_NON_QUARKUS_TEST);
+        this.secondaryOrderer = config.getOptionalValue(CFGKEY_SECONDARY_ORDERER, String.class);
+    }
+
+    QuarkusTestProfileAwareClassOrderer(
+            final String prefixQuarkusTest,
+            final String prefixQuarkusTestWithProfile,
+            final String prefixQuarkusTestWithRestrictedResource,
+            final String prefixNonQuarkusTest,
+            final Optional<String> secondaryOrderer) {
+        this.prefixQuarkusTest = prefixQuarkusTest;
+        this.prefixQuarkusTestWithProfile = prefixQuarkusTestWithProfile;
+        this.prefixQuarkusTestWithRestrictedResource = prefixQuarkusTestWithRestrictedResource;
+        this.prefixNonQuarkusTest = prefixNonQuarkusTest;
+        this.secondaryOrderer = secondaryOrderer;
+    }
+
     @Override
     public void orderClasses(ClassOrdererContext context) {
         // don't do anything if there is just one test class or the current order request is for @Nested tests
         if (context.getClassDescriptors().size() <= 1 || context.getClassDescriptors().get(0).isAnnotated(Nested.class)) {
             return;
         }
-        var prefixQuarkusTest = getConfigParam(
-                CFGKEY_ORDER_PREFIX_QUARKUS_TEST,
-                DEFAULT_ORDER_PREFIX_QUARKUS_TEST,
-                context);
-        var prefixQuarkusTestWithProfile = getConfigParam(
-                CFGKEY_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE,
-                DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_PROFILE,
-                context);
-        var prefixQuarkusTestWithRestrictedResource = getConfigParam(
-                CFGKEY_ORDER_PREFIX_QUARKUS_TEST_WITH_RESTRICTED_RES,
-                DEFAULT_ORDER_PREFIX_QUARKUS_TEST_WITH_RESTRICTED_RES,
-                context);
-        var prefixNonQuarkusTest = getConfigParam(
-                CFGKEY_ORDER_PREFIX_NON_QUARKUS_TEST,
-                DEFAULT_ORDER_PREFIX_NON_QUARKUS_TEST,
-                context);
 
         // first pass: run secondary orderer first (!), which is easier than running it per "grouping"
-        buildSecondaryOrderer(context).orderClasses(context);
```

Comment:
AFAICS `buildSecondaryOrderer()` is not called anymore so it should be removed.

Reply:
Removed.